sdl2promela/translator.py

```diff
@@ -296,36 +386,498 @@ def __generate_statement(
+        if length < int(finalType.Min) or length > int(finalType.Max):
+            raise Exception(
+                f"Invalid assignment: {finalType.CName} does not accept string with length {length}"
+            )
+        length_field = (
+            MemberAccessBuilder()
+            .withUtypeReference(__generate_variable_name(sdl_model, left, True))
+            .withMember(VariableReferenceBuilder("length").build())
+            .build()
+        )
+        data_field = sdlmodel.MemberAccess(left, sdlmodel.VariableReference("data"))
+
+        statements: List[promelamodel.Statement] = []
+
+        if int(finalType.Min) != int(finalType.Max):
+            statements.append(
+                AssignmentBuilder()
+                .withTarget(length_field)
+                .withSource(promelamodel.IntegerValue(length))
+                .build()
+            )
+        for index in range(int(finalType.Max)):
+            if index < length:
+                value = promelamodel.IntegerValue(ord(right.value[index]))
+            else:
+                value = promelamodel.IntegerValue(0)
+            element = (
+                ArrayAccessBuilder()
+                .withArray(__generate_variable_name(sdl_model, data_field, True))
+                .withIndex(promelamodel.IntegerValue(index))
+                .build()
+            )
+            statements.append(
+                AssignmentBuilder().withTarget(element).withSource(value).build()
+            )
+
+        return statements
+    elif finalType.kind == "BitStringType":
+        raise NotImplementedError("Assignment to BIT STRING is not supported")
+    else:
+        raise Exception(f"Unsupported assignment: {finalType.kind} EmptyString")
+
+
+@dispatch(
+    sdlmodel.Model,
+    (
+        sdlmodel.VariableReference,
+        sdlmodel.ArrayAccess,
+        sdlmodel.MemberAccess,
+    ),
+    sdlmodel.OctetStringValue,
+    type,
+)
+def __generate_assignment(
+    sdl_model: sdlmodel.Model,
+    left: Union[
+        sdlmodel.VariableReference, sdlmodel.ArrayAccess, sdlmodel.MemberAccess
+    ],
+    right: sdlmodel.OctetStringValue,
+    left_type: type,
+) -> List[promelamodel.Statement]:
+    finalType = resolve_asn1_type(sdl_model.types, left_type)
+    if finalType.kind == "OctetStringType" or finalType.kind == "IA5StringType":
+        length = len(right.elements)
+        if length < int(finalType.Min) or length > int(finalType.Max):
+            raise Exception(
+                "Invalid assignment: {} does not accept string with length {}".format(
+                    finalType.CName, length
+                )
+            )
+
+        length_field = (
+            MemberAccessBuilder()
+            .withUtypeReference(__generate_variable_name(sdl_model, left, True))
+            .withMember(VariableReferenceBuilder("length").build())
+            .build()
+        )
+        data_field = sdlmodel.MemberAccess(left, sdlmodel.VariableReference("data"))
```

Comment:
Consider creating "constants" for strings used multiple times, like data or length -> these could be potentially changed later

Reply:
Done.